servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -1087,6 +1088,32 @@ public final Cancellable subscribe(Consumer<? super T> resultConsumer) {
         return new SucceededSingle<>(value);
     }
 
+    /**
+     * Creates a realized {@link Single} which always completes successfully with the result of the provided
+     * {@code callable}.
+     *
+     * @param callable {@link Callable} which supplies the result of the {@link Single}.
+     * @param <T>      Type of the {@link Single}.
+     *
+     * @return A new {@link Single}.
+     */
+    public static <T> Single<T> fromCallable(final Callable<T> callable) {
+        return new CallableSingle<>(callable);
+    }
+
+    /**
+     * Creates a realized {@link Single} which always completes successfully with the result of the provided
+     * {@code supplier}.
+     *
+     * @param supplier {@link Supplier} which supplies the result of the {@link Single}.
+     * @param <T>      Type of the {@link Single}.
+     *
+     * @return A new {@link Single}.
+     */
+    public static <T> Single<T> fromSupplier(final Supplier<T> supplier) {
+        return new SupplierSingle<>(supplier);
```

Comment:
Do we need both `fromCallable` and `fromSupplier` methods? It seems like `fromCallable` should be sufficient (since `Callable#call()` accomodates for throwing an `Exception`) and if users have a `Supplier` they can use `fromCallable(supplier::get)`.

In case you think both methods are important to provide, we should reduce implementations by delegating to `fromCallble()` from here, like:

```java
public static <T> Single<T> fromSupplier(final Supplier<T> supplier) {
    return fromCallable(supplier::get);
}
```

Reply:
Sounds good!